### ISSUE TITLE
FileGDB support missing

### ISSUE THREAD

**Issue by grwhumphries** [Toblerity/Fiona#565]
Simply asking for FileGDB support to write out to ESRI GDB files.  Seems like it was around in older versions, but has disappeared in later versions.    

Otherwise, it would be great to have good documentation on how to enable the feature, or how to get around it on the fiona end as writing to ESRI GDBs is a commonly performed task in industry.

**Comment by micahcochran** [comment]
FileGDB requires GDAL to be compiled with ESRI's FileGDB API SDK.  See [GDAL docs for FileGDB under the section "Building Notes"](http://www.gdal.org/drv_filegdb.html) and see [this](http://trac.osgeo.org/gdal/wiki/FileGDB).

**Comment by grwhumphries** [comment]
Most of these instructions are jargon and mean very little to people who aren't developers. For those people who want to just do simple scripting (e.g., write and write to an ESRI GDB file in batches), it would be far better to have layman's instructions somewhere on how to do this from steps A to Z.  Lots of users out there who would benefit from this without having to learn to be a developer.

**Comment by micahcochran** [comment]
The problem here is there are many operating systems and many different ways to install GDAL and Fiona.  Compiling is a sure fire way to give instructions that allow anyone to customize their installation with whatever options they want.

What is your operating system?  Where did you get a your binaries for GDAL and Fiona?

**Comment by sgillies** [comment]
@grwhumphries It's my understanding that I can't redistribute Esri's FileGDB API SDK. You have to download it yourself and make a GDAL plugin. Once you have, it should only take a little bit of configuration (documented here: https://trac.osgeo.org/gdal/wiki/ConfigOptions#GDAL_DRIVER_PATH) to turn it on for Fiona.

I've read that a special arrangement has been made for OSGeo's OSGeo4W distribution, but I'm not sure that Fiona is included with that yet. Lobbying Python software distributions (like Anaconda, or Enthought, or OSGeo) to include the FileGDB plugin for GDAL/OGR is the thing to do for users who aren't developers.

**Comment by grwhumphries** [comment]
Thanks for clarification on the SDK @sgillies  - makes sense.     

the FileGDB drivers can be installed with osgeo4w as part of the setup, which is helpful, but to find that information, without any prior instruction, would be a nightmare for people with a basic knowledge of scripting. 

I just think it would be really helpful to have some clear instructions - a literal "point and click here and here" - without referring to the osgeo.org instructions, which are really cumbersome for non-developers - jargony and the like.   

Basically, you want someone to be able to "pip install fiona"  - then (for eaxmple).. download Osgeo4w with the FileGDB SDK through the setup menu - then set path variables (GDAL_DRIVER_PATH), etc...

**Comment by lpinner** [comment]
@sgillies I think your understanding of the Esri FileGDB API SDK license is a bit dated. Though the binary bits are not open source, they can still be redistributed freely as the SDK as a whole (i.e. the binaries, libs, docs and sample code) is [released under an Apache 2.0 license](https://github.com/Esri/file-geodatabase-api/blob/master/License.txt)

Anybody is free to redistribute all or part of the SDK.

This is probably more relevant to packagers though rather than fiona itself.

**Comment by awa5114** [comment]
Is there any progress on this?  This feature would be amazing to have...

**Comment by phildias** [comment]
I second this! I'm trying to use Geopandas 0.8 through Python 3.8 (installed using Anaconda) and I'm also still having a hard time getting this to work. The instructions found [here](https://gis.stackexchange.com/questions/193288/how-to-add-support-for-filegdb-esri-file-gdb-api-driver-in-fiona) (downloading FileGDBAPI.dll from ESRI's site, placing it in the appropriate folder and adding an environment variable) did not seem to work.

**Comment by phildias** [comment]
Is there any chance this can get added to the 1.9 milestone? 
Also, I might be able work on this myself, but will likely need a bit of direction, since I'm not super familiar with how fiona handles these external DLL drivers.

**Comment by rbuffat** [comment]
@phildias This is a packaging issue that only affects Fiona wheels listed on pypi (The *whl files listed here: https://pypi.org/project/Fiona/#files). If you use another package distribution, such as Anaconda, it depends only if the version of GDAL provided by Anaconda supports FileGDB or not. Thus, I would recommend directly asking the Anaconda project for FileGDB support for GDAL.

**Comment by phildias** [comment]
Thanks for the prompt reply! I can try that route. In the meantime, do you happen to know how to allow Fiona to use ESRI's FileGDB driver (the "FileGDBAPI.dll" files inside of [these zipfiles](https://github.com/Esri/file-geodatabase-api/)) after I've installed Fiona through Anaconda? I imagined it would be as easy as extracting the DLL files into GDAL's python installation folder and setting up a couple of environment variables, but that doesn't seem to be doing the trick.
